phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIndexIdRetrieveIT.java

```diff
@@ -0,0 +1,169 @@
+             final Statement stmt = conn.createStatement()) {
+            stmt.execute(String.format(ddlForBaseTable, fullTableName));
+            stmt.execute(String.format(ddlForView, viewFullName, fullTableName));
+            stmt.execute(String.format(ddlForViewIndex, viewIndexName, viewFullName));
+
+            ResultSet rs = stmt.executeQuery(String.format(selectRow, viewIndexName));
+            rs.next();
+            assertEquals(Short.MIN_VALUE,rs.getLong(1));
+            assertEquals(expectedDataType, rs.getInt(2));
+            assertFalse(rs.next());
+        }
+    }
+
+    @Test
+    public void testMixedCase() throws Exception {
+        // mixed case
+        Properties propsForLongType = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        propsForLongType.setProperty(LONG_VIEW_INDEX_ENABLED_ATTRIB, "true");
+        String schema = generateUniqueName();
+        String baseTable = generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(schema, baseTable);
+        String viewName = generateUniqueName();
+        String viewFullName = SchemaUtil.getTableName(schema, viewName);
+        String viewIndexName1 = generateUniqueName();
+
+        // view index id data type is long
+        try (final Connection conn = DriverManager.getConnection(url,propsForLongType);
+             final Statement stmt = conn.createStatement()) {
+            stmt.execute(String.format(ddlForBaseTable, fullTableName));
+            stmt.execute(String.format(ddlForView, viewFullName, fullTableName));
+            stmt.execute(String.format(ddlForViewIndex, viewIndexName1, viewFullName));
+
+            ResultSet rs = stmt.executeQuery(String.format(selectRow, viewIndexName1));
+            rs.next();
+            assertEquals(Short.MIN_VALUE,rs.getLong(1));
+            assertEquals(Types.BIGINT, rs.getInt(2));
+            assertFalse(rs.next());
+        }
+
+        // view index id data type is short
+        String viewIndexName2 = generateUniqueName();
+        Properties propsForShortType = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        propsForShortType.setProperty(LONG_VIEW_INDEX_ENABLED_ATTRIB, "false");
+        try (final Connection conn = DriverManager.getConnection(url,propsForShortType);
+             final Statement stmt = conn.createStatement()) {
+            stmt.execute(String.format(ddlForViewIndex, viewIndexName2, viewFullName));
+
+            ResultSet rs = stmt.executeQuery(String.format(selectRow, viewIndexName2));
+            rs.next();
+            assertEquals(Short.MIN_VALUE + 1,rs.getLong(1));
+            assertEquals(Types.SMALLINT, rs.getInt(2));
+            assertFalse(rs.next());
+        }
+
+        // check select * from syscat
+        try (final Connection conn = DriverManager.getConnection(url);
+             final Statement stmt = conn.createStatement()) {
+            ResultSet rs = stmt.executeQuery(String.format(selectAll));
```

Comment:
Why not just query for the exact  metadata rows for the view indices instead of select * .

Reply:
because we want to test the mixed cases.
in this test, we create two views (one with smallint config, and one with bigint config)